api/graphql/reservations/reservation_serializers/mixins.py

```diff
@@ -306,9 +306,11 @@ def check_reservation_duration(self, reservation_unit: ReservationUnit, begin, e
             )
 
     def check_buffer_times(self, reservation_unit, begin, end):

```

Comment:
Got me pondering should we add the `if ResrevationType.BLOCKED` to here to have the iffing only one place 🤔 
On the other hand its more flexible as is now so that the buffers can be checked to blocked ones in some serializers although I don't see the need and that could be arranged if needed. 

So whatya think?

Reply:
I added an optional `reservation_type` parameter which can be given in `CREATE` cases where `self.instance` does not exist, and then then I moved the `if` statement inside `check_buffer_times`. Simplified the code quite nicely 👍